lib/controllers/adm/services.ts

```diff
@@ -70,13 +66,13 @@ type IUpdateServiceHandler = (
  *
  * TODO: validate the payload against a schema.
  */
-const ServicePayloadMiddleware: IRequestMiddleware<
+export const ServicePayloadMiddleware: IRequestMiddleware<
   IResponseErrorValidation,
-  ISerializableService
+  IService
 > = request => {
   const body = request.body;
-  const servicePayload: ISerializableService = {
-    authorizedRecipients: body.authorized_recipients,
+  const servicePayload: IService = {
+    authorizedRecipients: toAuthorizedRecipientsSet(body.authorized_recipients),
```

Comment:
avoid leaking the data structure to the method, if tomorrow we're going to use a different data structure than a Set, we don't want to change the method name... `toAuthorizedRecipients` would be better